src/analyses/constant_propagator.cpp

```diff
@@ -35,8 +36,7 @@ void constant_propagator_domaint::assign_rec(
   const symbol_exprt &s=to_symbol_expr(lhs);
 
   exprt tmp=rhs;
-  values.replace_const(tmp);
-  simplify(tmp, ns);
+  try_evaluate(tmp, ns);
```

Comment:
Isn't what you are doing precisely what's known as "partial evaluation"? The "try_" prefix, to me, doesn't convey the intent properly.

Reply:
The expression evaluation here is partial, but if the result can't be reduced to a single constant the evaluation fails in the constant propagator domain so I thought the name was appropriate. But it's bad if the name is unclear, what would you like it to be called?

Reply:
Invoking `try_evaluate(tmp, ns)` modifies `tmp` based on whatever information is available in the current context, so I'm not sure I agree it "fails?" It may of course be the case that the result isn't a constant and thus isn't stored, but partial evaluation has been performed.

Reply:
In my opinion it would be `partial_evaluate`  or `partial_evaluation`.